io/net/virt-net/network_virtualization.py

```diff
@@ -420,6 +423,44 @@ def test_vnic_auto_failover(self):
         else:
             self.cancel("Provide more backing device, only 1 given")
 
+    def test_rmdev_viosfailover(self):
+        '''
+        using mrdev and mkdev command to check vios failover
+        '''
+        self.login(self.vios_ip, self.vios_user, self.vios_pwd)
```

Comment:
Can you handle the vios session and hmc session using separate objects ?

Reply:
run_command uses class parameter self.pxsh for console. this also needs to be modified
if separate obj for console, than run_command should be supplied with the console obj each time where it is called

Reply:
@narasimhan-v can you elaborate more here ?

Reply:
@abdhaleegit yes, we need to call run_command with the separate object, since we are handling sessions for 2 different systems here (hmc and vios). Also, when we support multiple vios, we need separate objects for each one of them.

Reply:
we can't call run_command through object.

Reply:
@smruti77 we should pass object as an argument to `run_command()`

Reply:
login():
    get login session
    return pxh   (instead of using self.pxssh)
....
.....
run_command(self, cmd, console):
    console.sendline(cmd)  (instead of self.pxsh.sendlie)
    .....
    .....
when calling
con_hmc = login(hmc,...)
run_command(cmd, con_hmc)
...
con_vios1 = login(vios..)
run_command(cmd, con_vios1)``